snakemake/executors/__init__.py

```diff
@@ -788,11 +788,11 @@ def format_job(self, pattern, job, **kwargs):
 
             waitfiles_parameter = format(
                 "--wait-for-files-file {wait_for_files_file}",
-                wait_for_files_file=wait_for_files_file,
+                wait_for_files_file=('"' + wait_for_files_file + '"'),
             )
         else:
             waitfiles_parameter = format(
-                "--wait-for-files {wait_for_files}", wait_for_files=wait_for_files
+                "--wait-for-files {wait_for_files}", wait_for_files=(" ".join(['"{0}"'.format(x) for x in wait_for_files])),
```

Comment:
Same as above.

Reply:
You need this: `wait_for_files=[repr(f) for f in wait_for_files]`, i.e., apply it to each item of the list, not the entire list. Otherwise you get the representation of the entire list as a single string, this is not what you want.

Reply:
Hi Johannes,

Thank you  for your guidance and patience. I definitely learned something. I apply the `repr()` on both section and tested the new version. Works like a charm. PR is now updated.

Thank you again,

Fred

Reply:
Great! Now it just needs a final black reformatting :-).

Reply:
Done! (did not know about black reformatting but this is pretty handy. Thanks.)